includes/utilities.inc

```diff
@@ -143,3 +140,61 @@ function islandora_ead_get_level_info(AbstractObject $ead, $refid) {
   $level_info = implode(', ', $containers);
   return $level_info;
 }
+
+/**
+ * Helper to transform XML using the named XSLT with saxonb.
+ *
+ * @param string $xsl_name
+ *   Name of the XSL file to be used, relative to this module's "xml"
+ *   directory.
+ * @param string $doc
+ *   The XML to convert.
+ * @param array $params
+ *   An optional associative array of XSLT parameters mapping parameter names
+ *   to their values.
+ *
+ * @return string
+ *   The transformed xml.
+ */
+function islandora_ead_transform_with_saxonb($xsl_name, $doc, array $params = array()) {
+  $module_path = drupal_get_path('module', 'islandora_ead');
+  $xsl_path = format_string('!path/xml/!xsl', array(
+    '!path' => $module_path,
+    '!xsl' => $xsl_name,
+  ));
+
+  $pipes = array();
+  $equal_join_and_escape = function ($key, $value) {
+    return escapeshellarg("$key=$value");
+  };
+  $escaped = array_map($equal_join_and_escape, array_keys($params), $params);
+  $param_string = implode(' ', $escaped);
+
+  $process = proc_open("saxonb-xslt -versionmsg:off - $xsl_path $param_string",
+    array(
+      0 => array(
+        'pipe',
+        'r',
+      ),
+      1 => array(
+        'pipe',
+        'w',
+      ),
+      2 => array(
+        'file',
+        '/dev/null',
+        'a',
+      ),
+    ),
+    $pipes
+  );
+
+  fwrite($pipes[0], $doc);
+  fclose($pipes[0]);
+  $saxonb_transformed = stream_get_contents($pipes[1]);
+  fclose($pipes[1]);
+  proc_close($process);
+
+  watchdog('islandora_ead', $saxonb_transformed);
```

Comment:
Watchdogging the output for any particular reason? Also, feels like we may want the contents of pipe 2 rather than chuckin' it?

Reply:
No particular reason, it's just because this function is cribbed from bhofdora